src/index.d.ts

```diff
@@ -0,0 +1,29 @@
+import { ReactNode, FC } from 'react';
+
+export interface ExternallyChangedProps {
+  name: string;
+  children: (externallyChanged: boolean) => ReactNode;
+}
+
+export interface OnBlurProps {
+  name: string;
+  children: () => void;
+}
+
+export interface OnChangeProps {
+  name: string;
```

Comment:
i'd suggest to set up types here more strongly than just `name: string`.
It might be stated as [keyof](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html) of some model at least.
Ideally should support nested names like `name="model.address.city"`.

Reply:
> Thank you for the review @ktarasov-clearscale ! It's actually a nice point. But I can't find a good solution so far. I've tried to use generics in order to make it available for the user to pass an interface. But it didn't work. Here's what I tried:
> 
> ```
> export type OnBlurComponent<T> = FC<OnBlurProps<T>>;
> export const OnBlur: OnBlurComponent;
> ```
> 
> I see people actually struggle with generics on FC, StackOverflow has a bunch of discussions around this topic. So if you know how to fix it please let me know... Thanks again!

Here is my suggestion for full type safety:
```
interface OnChangePropsName<T> {
    name: keyof T;
}
interface OnChangeProps<T> extends OnChangePropsName<T>{
    children: (value: T[OnChangePropsName<T>['name']], previous: T[OnChangePropsName<T>['name']]) => void;
}

type OnChange<T> = React.ComponentType<OnChangeProps<T>>;
```

Reply:
@sorokin-evgeni the solution you've suggested doesn't work for me, unfortunately. Did you test it?

Reply:
Sure, it works correctly in my project.
Can you give me code sample and error text please?

Reply:
It was complaining to me that in this way `OnChange` represents a type, but used as a value.

Reply:
I don't know for now how to infer type of value in OnChange function. But type of name is working correctly in my case:
```
interface OnChangeProps<T> {
    name: keyof T;
    children: (
      value: any, 
      previous: any
      ) => void;
}

export class OnChange<T> extends Component<OnChangeProps<T>> {};
```

Reply:
@sorokin-evgeni I got your idea, thanks! Could you please also share the example of usage `<OnChange />` component in this case? More specifically, how do you pass the type/interface to it?